main.tf

```diff
@@ -14,15 +14,54 @@
  * limitations under the License.
  */
 
+provider "google" {
+  version = "~> 3.0"
+}
+
+data "google_client_config" "current" {
+}
+
+
+module "data_fusion_network" {
+  source          = "./modules/private_data_fusion_network"
+  project_id      = var.project
+  tenant_project  = module.instance.tenant_project
+  instance        = module.instance.instance.name
+  vpc_network     = var.network
+  dataproc_subnet = var.dataproc_subnet
+  region          = var.region
+}
+
 module "instance" {
   source = "./modules/instance"
 
-  name           = var.name
-  project        = var.project
-  description    = var.description
-  region         = var.region
-  type           = var.type
-  labels         = var.labels
-  options        = var.options
-  network_config = var.network_config
+  name        = var.name
+  project     = var.project
+  description = var.description
+  region      = var.region
+  type        = var.type
+  labels      = var.labels
+  options     = var.options
+  network_config = {
+    network       = module.data_fusion_network.data_fusion_vpc.name
+    ip_allocation = module.data_fusion_network.data_fusion_ip_allocation
+  }
+}
+
+provider "cdap" {
+  host  = "${module.instance.instance.service_endpoint}/api/"
+  token = data.google_client_config.current.access_token
+}
+
+module "verbose_dataproc" {
```

Comment:
Is there a reason the dataproc is being created in the main module? I think it would be better if any CDAP resources were in separate modules just so the main module can be deployed without needing to install the cdap provider

Reply:
good point.